src/client/Session.cpp

```diff
@@ -35,11 +61,10 @@ bool Session::ping() {
 }
 
 ErrorCode Session::retryConnect() {
-    pool_->giveBack(std::move(conn_));
-    conn_ = pool_->getConnection();
-    auto resp = conn_.authenticate(username_, password_);
-    sessionId_ = resp.sessionId != nullptr ? *resp.sessionId : -1;
-    return resp.errorCode;
+    if (!conn_.open()) {
```

Comment:
retry connect means to get an ok connection from the pool, no means to reopen the broken。

Reply:
Fixed.